PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskPID.cxx

```diff
@@ -1270,12 +1270,12 @@ void AliAnalysisTaskPID::UserCreateOutputObjects()
   }
   
   if (GetDoTPCclusterStudies()) {
-    Int_t nBinsPrime = 320;
-    Double_t primeLow = 0.4;
-    Double_t primeHigh = 2.0;
-    Int_t arrNumbers = 4;
+    const Int_t nBinsPrime = 320;
+    const Double_t primeLow = 0.4;
+    const Double_t primeHigh = 2.0;
+    const Int_t arrNumbers = 4;
     fTPCclusterStudies = new TObjArray*[arrNumbers];
-    Int_t histNumbers = 3;
+    const Int_t histNumbers = 3;
     
     const Int_t nBinsY[histNumbers+1] = {8, 4, 11, 15};
```

Comment:
The problem is here: you declare the array as const Int, but declaring an array `const type` doesn't mean it is actually initialised. So the compiler thinks the array might be uninitialised and considers variable size array length in the arrays you use to these values to define the sizes. Even though you initialised the array the compiler will not evaluate this at compile time.

In order to circumvent such problems c++11 comes with a new keyword `constexpr`, and since the line is in the cxx file you are free to use it as all our builds require at least c++11. `constexpr` forces the compiler to evaluate a statement at compile time and use the result as a const in all places in the code. So changing the line to

```c++
constexpr Int_t nBinsY[histNumbers+1] = {8, 4, 11, 15};
```

will work.

Reply:
Thanks for the suggestion. I have used constexpr now